tests/director/tests/formation_api_test.go

```diff
@@ -1978,102 +2083,286 @@ func TestFormationAssignments(stdT *testing.T) {
+			assertNotificationsCountForTenant(t, body, localTenantID, 2)
+			notificationsForConsumerTenant = gjson.GetBytes(body, localTenantID)
+			assertExpectationsForApplicationNotifications(t, notificationsForConsumerTenant.Array(), []*applicationFormationExpectations{
+				{
+					op:                 assignOperation,
+					formationID:        formation.ID,
+					objectID:           rtCtx.ID,
+					subscribedTenantID: rtCtx.Value,
+					objectRegion:       regionLbl,
+					configuration:      "",
+				},
+				{
+					op:                 assignOperation,
+					formationID:        formation.ID,
+					objectID:           rtCtx.ID,
+					subscribedTenantID: rtCtx.Value,
+					objectRegion:       regionLbl,
+					configuration:      "{\"asyncKey\":\"asyncValue\",\"asyncKey2\":{\"asyncNestedKey\":\"asyncNestedValue\"}}",
+				},
+			})
+
+			var unassignFormation graphql.Formation
+			t.Logf("Unassign application from formation %s", formation.Name)
+			unassignReq := fixtures.FixUnassignFormationRequest(actualApp.ID, string(graphql.FormationObjectTypeApplication), providerFormationName)
+			err = testctx.Tc.RunOperationWithCustomTenant(ctx, certSecuredGraphQLClient, subscriptionConsumerAccountID, unassignReq, &unassignFormation)
+			require.NoError(t, err)
+			require.Equal(t, formation.Name, assignedFormation.Name)
+
+			application := fixtures.GetApplication(t, ctx, certSecuredGraphQLClient, subscriptionConsumerAccountID, actualApp.ID)
+			scenarios, hasScenarios := application.Labels["scenarios"]
+			assert.True(t, hasScenarios)
+			assert.Len(t, scenarios, 1)
+			assert.Contains(t, scenarios, providerFormationName)
+
+			body = getNotificationsFromExternalSvcMock(t, certSecuredHTTPClient)
+
+			// rtCtx <- App notifications
+			assertNotificationsCountForTenant(t, body, subscriptionConsumerTenantID, 2)
+			notificationsForConsumerTenant = gjson.GetBytes(body, subscriptionConsumerTenantID)
+
+			unassignNotificationFound := false
+			for _, notification := range notificationsForConsumerTenant.Array() {
+				op := notification.Get("Operation").String()
+				if op == unassignOperation {
+					appIDFromNotification := notification.Get("RequestBody.items.0.ucl-system-tenant-id").String()
+					t.Logf("Found notification for app %q", appIDFromNotification)
+					if appIDFromNotification == actualApp.ID {
+						unassignNotificationFound = true
+						assertFormationNotification(t, notification, unassignOperation, formation.ID, actualApp.ID, localTenantID, appNamespace, appRegion)
+					}
 				}
 			}
-		}
-		require.True(t, unassignNotificationFound)
+			require.True(t, unassignNotificationFound)
+
+			// rtCtx -> App notifications
+			assertNotificationsCountForTenant(t, body, localTenantID, 3)
+			notificationsForConsumerTenant = gjson.GetBytes(body, localTenantID)
+			assertSeveralFormationNotifications(t, notificationsForConsumerTenant, rtCtx, formation.ID, regionLbl, unassignOperation, 1)
+
+			//list assignments - expect 0
```

Comment:
expect 1, but I would propose to remove those comments. Those are the kind of comments that exactly replicate the semantics of the code and it is inevitable to become outadate and therefore making more harm than good.